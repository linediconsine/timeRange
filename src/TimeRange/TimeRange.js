import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import './TimeRange.css';
const TimeRange = ({ startTime, endTime, ranges }) => {
    const width = 500;
    const height = 3;
    const offesetX = 10;
    const offesetYtext = 20;
    const offesetY = 10;
    console.log(startTime, endTime, width);
    return (_jsxs("svg", { width: width + 10, height: height + offesetY + offesetYtext, children: [_jsx("rect", { className: "base", x: offesetX / 2, y: offesetY / 2 + offesetYtext, width: width, height: height, fill: "#f0f0f011" }), ranges?.map((range, index) => {
                console.log(range.start, range.end);
                const pxStart = width * range.start / endTime;
                const pxWidth = width * (range.end - range.start) / endTime;
                return _jsxs("g", { fill: "#007bff", className: `range ${range.class}`, children: [_jsx("text", { className: 'text', x: pxStart + offesetX / 2, y: offesetYtext / 2 + 5, fill: "red", children: range.start }), _jsx("text", { className: 'text', x: pxStart + pxWidth - offesetX / 2, y: offesetYtext / 2 + 5, fill: "red", children: range.end }), _jsx("circle", { className: "startRange", cx: pxStart + offesetX / 2, cy: offesetY / 2 + 2 + offesetYtext, r: 4 }), _jsx("rect", { x: pxStart + 5, y: 5 + offesetYtext, width: pxWidth, height: height, "data-title": "Time Range " + range.start + " to " + range.end }, index), _jsx("circle", { className: "endRange", cx: pxStart + pxWidth + offesetX / 2, cy: offesetY / 2 + 2 + offesetYtext, r: 4 })] });
            })] }));
};
export default TimeRange;
