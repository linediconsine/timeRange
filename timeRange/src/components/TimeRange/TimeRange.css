.base {
    stroke-width: 1;
    stroke: rgb(0, 0, 0)
}

.loading {
    animation: pulse 2s infinite;
}

@keyframes pulse {

    0%,
    100% {
        fill: #8b9eff;
    }

    50% {
        fill: #0c32af;
    }
}