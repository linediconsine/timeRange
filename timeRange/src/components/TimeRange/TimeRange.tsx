import React from 'react';

interface TimeRangeProps {
    startTime: number;
    endTime: number;
    ranges?: { start: number; end: number }[];
}

const TimeRange: React.FC<TimeRangeProps> = ({ startTime, endTime, ranges }) => {
    const width = 500;
    const height = 3;
    const offesetX = 10;
    
    const offesetYtext = 20;
    const offesetY = 10;

    console.log(startTime, endTime, width)
    return (
        <svg width={width + 10} height={height + offesetY + offesetYtext}>
            <rect className="base" x={offesetX/2} y={offesetY/2 + offesetYtext} width={width} height={height} fill="#f0f0f011" />

            {ranges?.map((range, index) => {
                console.log(range.start, range.end);

                const pxStart = width * range.start / endTime;
                const pxWidth = width * (range.end - range.start) / endTime;


                return <g fill="#007bff" className="range" >
                    <text className='text' x={pxStart + offesetX/2} y={offesetYtext/2 + 5} fill="red">{range.start}</text>
                    <text className='text' x={pxStart + pxWidth - offesetX/2 } y={offesetYtext/2 + 5} fill="red">{range.end}</text>

                    <circle cx={pxStart + offesetX/2} cy={offesetY/2 + 2 +   offesetYtext} r={4} />
                    <rect
                        key={index}
                        x={pxStart + 5}
                        y={5 +  offesetYtext}
                        width={pxWidth}
                        height={height}
                        data-title={"Time Range " + range.start + " to " + range.end}
                    />
                    <circle cx={pxStart + pxWidth + offesetX/2 } cy={offesetY/2 + 2 +   offesetYtext} r={4} />
                </g>

            })}

        </svg>
    );
};

export default TimeRange;