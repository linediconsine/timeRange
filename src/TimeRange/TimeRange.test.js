import { jsx as _jsx } from "react/jsx-runtime";
import { expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import TimeRange from './TimeRange';
it('renders correctly', () => {
    const ranges = [
        { start: 0, end: 2 },
        { start: 100, end: 150 },
        { start: 20, end: 25 },
        { start: 25, end: 30 },
    ];
    render(_jsx(TimeRange, { startTime: 0, endTime: 30, ranges: ranges }));
    const result = screen.debug();
    expect(result).toMatchSnapshot();
});
