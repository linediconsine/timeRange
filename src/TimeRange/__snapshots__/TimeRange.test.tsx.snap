// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`end renders correctly 1`] = `undefined`;

exports[`renders correctly 1`] = `undefined`;

exports[`renders correctly 2`] = `undefined`;

exports[`renders correctly 3`] = `undefined`;

exports[`start renders correctly 1`] = `undefined`;
